jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/MassIndexingJobAvailableParameters.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.jsr352.massindexing;
+
+
+/**
+ * @author Yoann Rodiere
+ */
+public final class MassIndexingJobAvailableParameters {
```

Comment:
I totally agree with you, Yoann. Thanks for your detailed explanation. In order to make sure the understanding, I made a summary here:

1. **Use an explicit name for utility class:** Differentiate using only the package name is bad, in particular in IDE auto-completion.
2. **The `Available` keyword could be removed:** it was set to be aligned with the Hibernate naming style, but it could be removed to shorten the class name.
3. **Job parameters and job properties are different terminologies in JSR-352**.
  a. Job parameter are provided by user via job operator (external configuration).
  b. Job properties are injected and used in batch artifacts (internal usage).
4. **The most important thing is public API**, i.e. how to ensure these job parameters are  meaningful and easy enough to be used. There's a shortcut between job parameters and job properties, because they have the exact same names in our case. <sup>(1)</sup>
5. **Plural is generally preferred for utility class,** because it centralizes different constant definitions (string here). Singular style is more likely to be used as a type, or enum.

<sup>(1)</sup> IMHO, it's good the current "shortcut", no need to increase the complexity be introducing an internal key mapping between job properties and job parameters.